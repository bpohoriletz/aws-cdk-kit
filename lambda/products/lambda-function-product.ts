import * as lambda from 'aws-cdk-lib/aws-lambda';

export class LambdaFunctionProduct implements lambda.FunctionProps {
  runtime: lambda.Runtime;
  code: lambda.Code;
  handler: string;
}

//interface ILambdaFunctionProduct extends lambda.FunctionProps {}

export interface ILambdaFunctionBuilder {
  setRuntime(runtime: lambda.Runtime): ILambdaFunctionBuilder;
  setCode(code: lambda.Code): ILambdaFunctionBuilder;
  setHandler(handler: string): ILambdaFunctionBuilder;
  getResult(): lambda.FunctionProps;
}

export interface ILambdaFunctionBuilderConstructor {
  new (): ILambdaFunctionBuilder;
}
