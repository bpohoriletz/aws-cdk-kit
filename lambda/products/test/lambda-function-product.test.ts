import { LambdaFunctionProduct } from '../lambda-function-product';

describe('new LambdaFunctionProduct()', () => {
  it('has correct defaults', () => {
    expect(new LambdaFunctionProduct()).toMatchSnapshot();
  });
});
