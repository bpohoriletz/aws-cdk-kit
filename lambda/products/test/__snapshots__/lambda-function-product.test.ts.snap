// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new LambdaFunctionProduct() has correct defaults 1`] = `LambdaFunctionProduct {}`;
